django/cantusdb_project/main_app/views/chant.py

```diff
@@ -716,6 +692,32 @@ def get_context_data(self, **kwargs) -> dict:
         return context
 
     def get_queryset(self) -> QuerySet:
+        values_for_rendering_template = (
```

Comment:
Typically for static lists like this I define them somewhere external to the method itself, since there is no need to re-allocate the list every time the method is called. In itself, it would have a negligible effect on performance or memory usage, but I also find that just thinking in that way helps me to make lots of coding decisions that, in combination, makes a difference.

I think this list is also repeated below as well, so that also makes it a target for refactoring.

Reply:
> This list is also repeated below as well

Yes, it's repeated exactly, because both views use the same template. I thought of putting the list at the top of the file to avoid this duplication, but I expect to be making similar-but-different lists for other views (e.g. the FullIndexView needs a similar update). Better, I think, to put them in a consistent place near to where they're needed.

> somewhere external to the method itself, since there is no need to re-allocate the list every time the method is called

Django docs [suggest](https://docs.djangoproject.com/en/4.1/topics/class-based-views/intro/#using-class-based-views) that class-based views are "instantiated for each request dispatched to [them]", so defining this list in the `get_context_data` method will take just as much time/memory as if it's defined as a class variable. And since each view will have its own unique list of values, I think it makes most sense to define the list somewhere in the class.

Does this reasoning make sense? Is there a way to approach this I'm not thinking of?

Reply:
For lists that are almost similar, I typically define a core set of fields common to both, and then append the specific differences.

```
FOO = ["1", "2", "3"]
SPECIAL_FOO = [*FOO, "5"]
SPECIAL_BAR = [*FOO, "4"]
```

Often, I'll define these sorts of things in the top level of the file directly. That way they only get allocated once when the application is loaded. 

Again, the difference in computing resources for this specific instance is miniscule, but it helps to develop patterns like this as a way of working.

Reply:
Sorry, could have been clearer - similar in concept, but potentially quite different in terms of the values in the list.

Having sat with the idea for a while, I think a reasonable implementation is to include a list (or tuple, as the case may be) at the top of the file for the specific template .html file, and we can add additional similar-in-concept lists when we do similar things to different views. I'll make this change - thanks for pointing this out!